From 7db6a98b2bdcb0e11d3fdc8e78ea472725aafc32 Mon Sep 17 00:00:00 2001
From: Venu Vadapalli <vvadapalli@netlogicmicro.com>
Date: Fri, 10 Sep 2010 00:04:50 -0700
Subject: [PATCH 132/565] Fix for reading cpu's VC

Fix for reading cpu's VC (range is 0...3), export FDT to modules

Based on Broadcom SDK 2.3.

Signed-off-by: Venu Vadapalli <vvadapalli@netlogicmicro.com>
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 arch/mips/netlogic/xlp/on_chip.c | 5 +----
 arch/mips/netlogic/xlp/setup.c   | 5 +++--
 2 files changed, 4 insertions(+), 6 deletions(-)

diff --git a/arch/mips/netlogic/xlp/on_chip.c b/arch/mips/netlogic/xlp/on_chip.c
index 854fc2a..ca2db39 100644
--- a/arch/mips/netlogic/xlp/on_chip.c
+++ b/arch/mips/netlogic/xlp/on_chip.c
@@ -178,7 +178,6 @@ struct msgstn_handler msg_handler_map[XLP_MSG_HANDLE_MAX] = {
 void nlm_xlp_msgring_int_handler(unsigned int irq, struct pt_regs *regs)
 {
 	unsigned long mflags;
-	int vcpu;
 	int vc = 0;
 	uint32_t size = 0, code = 0, src_id = 0, cycles = 0;
 	struct msgstn_handler *handler = 0;
@@ -199,8 +198,6 @@ void nlm_xlp_msgring_int_handler(unsigned int irq, struct pt_regs *regs)
 
 	irq_enter();
 
-	vcpu = hard_smp_processor_id() & 0x1f;
-	vcpu = vcpu << 2;
         msgrng_access_enable(mflags);
 	cycles = read_c0_count();
 
@@ -211,7 +208,7 @@ void nlm_xlp_msgring_int_handler(unsigned int irq, struct pt_regs *regs)
 		vc_empty_status = (msg_status1 >> 24) & 0xf;
 		if (vc_empty_status == 0xf) break;
 
-		for( vc = vcpu; vc < (vcpu + 4); vc++)
+		for( vc = 0; vc < 4; vc++)
 		{
 			status = xlp_message_receive( vc, &src_id, &size, &code, &msg0, &msg1, &msg2, &msg3);
 			if(status != 0)
diff --git a/arch/mips/netlogic/xlp/setup.c b/arch/mips/netlogic/xlp/setup.c
index 97ead89..670bebf 100644
--- a/arch/mips/netlogic/xlp/setup.c
+++ b/arch/mips/netlogic/xlp/setup.c
@@ -92,6 +92,9 @@ unsigned long nlm_common_ebase = 0x0;
 static char prop_buf[MAX_PROP_LEN];
 extern char _end;
 extern void *fdt;
+
+EXPORT_SYMBOL(dt_ops);
+
 extern void *fdt_init(void *blob);
 extern void *simple_alloc_init(char *base, unsigned long heap_size,
 		unsigned long granularity, unsigned long max_allocs);
@@ -124,12 +127,10 @@ __u8 nlm_common_base_mac_addr[6];
 int chip_is_xls = 0;
 int chip_is_xls_b0 = 0;
 
-int xlp_sgmii_ports = 1;
 int hwemul = 0;
 
 EXPORT_SYMBOL(chip_is_xls);
 EXPORT_SYMBOL(chip_is_xls_b0);
-EXPORT_SYMBOL(xlp_sgmii_ports);
 EXPORT_SYMBOL(hwemul);
 
 void *nlm_common_psb_shm = 0;
-- 
1.8.4.93.g57e4c17

